src/support/small_vector.h

```diff
@@ -148,9 +148,11 @@ template<typename T, size_t N> class SmallVector {
   // iteration
 
   template<typename Parent, typename Iterator> struct IteratorBase {
+    using iterator_category = std::random_access_iterator_tag;
```

Comment:
It looks like there are still a bunch of operations missing to make this a valid random access iterator. See https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator for the docs.

Reply:
Thanks, I added a TODO for this.